### DISCUSSION TITLE
[Suggestions] Adventure map, hero should be able to land on any square where he can go by land

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#4724] [Ideas]
I suggest that the hero be able to land on any square where he can go by land.
The hero can't land on these squares in the original game either, but I think he should be able to do so in fheroes2 because I've noticed while watching let's plays that players don't understand and are surprised (and sometimes "angry", they also say, "there's a bug!") :) that they can't do it for no apparent reason.
This could be an option, so that the most resistant to the change from the original game would be happy. :)

In original game:

It is not possible to land just above the center of the town.
![Boat no move near land H2 002](https://user-images.githubusercontent.com/43583994/143785426-a108c740-6cac-4137-8729-d98d1bc588b1.png)

In fheroes2:

It is not possible to land just above the center of the town.
![Boat no move near land no direct landing 02](https://user-images.githubusercontent.com/43583994/143785487-639dfa82-e9fb-4cbc-b139-ca93f98b7b89.png)

Whereas nothing prevents the hero to go there when he is already on the land.
![Boat no move near land no direct landing 05](https://user-images.githubusercontent.com/43583994/143785485-89220276-244b-4134-8eca-5ca906a8e51f.png)

In original game:

There is a hero in this square at the beginning of the scenario.
![Boat no landing 001](https://user-images.githubusercontent.com/43583994/143851777-29c45d75-dfa7-4154-8d60-8cff14612e1a.png)

This prevents the hero from landing in this square
![Boat no landing 003](https://user-images.githubusercontent.com/43583994/143851809-a0f6386c-c3d3-4f64-9642-85693b66ec61.png)

There is no object in the square north of the southernmost Windmill at the beginning of the scenario.
![Boat no landing 002](https://user-images.githubusercontent.com/43583994/143851795-498f5a00-53db-45bb-8f00-805ab3107615.png)

However, the hero cannot land in this square.
![Boat no landing 004](https://user-images.githubusercontent.com/43583994/143851822-f749c215-c325-4d0a-b33b-ecbe9d73921a.png)

Although the hero can go to this square by land.
![Boat no landing 005](https://user-images.githubusercontent.com/43583994/143851836-322f020f-4800-46f5-abaa-72a047b6bd10.png)

- The hero cannot disembark because there is an object in this square at the beginning of the scenario (resource stack, wandering creatures, treasure chest, event, artifact, etc.).

When the object has been removed, the square is not marked as free.
The hero should also be able to land on a square where an event is located and trigger it.

This bug is also present in the original game.
There is a scenario to test this here: https://heroes2.forumactif.com/t851-heroes-2-bugs-blocked-disembarkment.

![Blocked disembarkment fh2 01](https://user-images.githubusercontent.com/43583994/97716186-9c68f080-1ac3-11eb-87a1-4a71b5cd2458.png)

_Originally posted by @LeHerosInconnu in https://github.com/ihhub/fheroes2/issues/2128#issuecomment-719582918_

- There is also a problem when the top of the sprite of a castle is close to the place where the hero wants to land although there is no object in this square at the beginning of the scenario.

In original game:

![Unable to land 02](https://user-images.githubusercontent.com/43583994/97804060-b8989900-1c4d-11eb-835a-700a76753214.png)

In fheroes2:

![Unable to land fh2 01](https://user-images.githubusercontent.com/43583994/97804062-be8e7a00-1c4d-11eb-840a-02e809c29294.png)

There are scenarios to test this here: https://heroes2.forumactif.com/t854-heroes-2-bugs-unable-to-land.

_Originally posted by @LeHerosInconnu in https://github.com/ihhub/fheroes2/issues/2128#issuecomment-720088502_

Here are the save files:
[No landing direct move 01.zip](https://github.com/ihhub/fheroes2/files/7614331/No.landing.direct.move.01.zip)
[Boat no landing 01.zip](https://github.com/ihhub/fheroes2/files/7617522/Boat.no.landing.01.zip)

**Comment by Branikolog** [comment]
Agree with @LeHerosInconnu. Game should be predictable: if we watch an empty coast, we should be able to land there. No matter, what is placed close to that place or what event was placed in editor.
